src/NzbDrone.Core/Download/Clients/Tribler/TriblerDownloadClient.cs

```diff
@@ -0,0 +1,359 @@
+            }
+        }
+
+        public override void RemoveItem(DownloadClientItem item, bool deleteData)
+        {
+            //_proxy.RemoveTorrent(item.DownloadId.ToLower(), deleteData, Settings);
+            var deleteDownloadRequestObject = new RemoveDownloadRequest
+            {
+                Remove_data = deleteData
+            };
+
+            var deleteRequestBuilder = getRequestBuilder("downloads/" + item.DownloadId.ToLower());
+            deleteRequestBuilder.Method = HttpMethod.DELETE;
+
+            // manually set content of delete request.
+            var deleteRequest = deleteRequestBuilder.Build();
+            deleteRequest.SetContent(Json.ToJson(deleteDownloadRequestObject));
+
+            ProcessRequest<DeleteDownloadResponse>(deleteRequest);
+        }
+
+        public override DownloadClientInfo GetStatus()
+        {
+            var config = getConfig();
+            var destDir = config.Settings.Download_defaults.Saveas;
+
+            if (Settings.TvCategory.IsNotNullOrWhiteSpace())
+            {
+                destDir = string.Format("{0}/.{1}", destDir, Settings.TvCategory);
+            }
+
+            return new DownloadClientInfo
+            {
+                IsLocalhost = Settings.Host == "127.0.0.1" || Settings.Host == "localhost",
+                OutputRootFolders = new List<OsPath> { _remotePathMappingService.RemapRemoteToLocal(Settings.Host, new OsPath(destDir)) }
+            };
+        }
+
+        protected override string AddFromMagnetLink(RemoteEpisode remoteEpisode, string hash, string magnetLink)
+        {
+            var addDownloadRequestObject = new AddDownloadRequest();
+            addDownloadRequestObject.Destination = GetDownloadDirectory();
+            addDownloadRequestObject.Uri = magnetLink;
+            addDownloadRequestObject.Safe_seeding = Settings.SafeSeeding;
+            addDownloadRequestObject.Anon_hops = Settings.AnonymityLevel;
+
+            // run hash through InfoHash class to ensure the correct casing.
+            var addDownloadRequestBuilder = getRequestBuilder("downloads");
+            addDownloadRequestBuilder.Method = HttpMethod.PUT;
+
+            // manually set content of delete request.
+            var addDownloadRequest = addDownloadRequestBuilder.Build();
+            addDownloadRequest.SetContent(Json.ToJson(addDownloadRequestObject));
+
+            string infoHashAsString = ProcessRequest<AddDownloadResponse>(addDownloadRequest).Infohash;
+
+            var infoHash = MonoTorrent.InfoHash.FromHex(infoHashAsString);
+            return infoHash.ToHex();
+        }
+
+        protected override string AddFromTorrentFile(RemoteEpisode remoteEpisode, string hash, string filename, byte[] fileContent)
+        {
+            // tribler api currently do not support recieving a torrent file direcly.
+            // instead extract infohash and generate a magnet uri.
+
+            var torrentFile = MonoTorrent.Torrent.Load(fileContent);
```

Comment:
If there isn't a torrent file I don't think this code should ever be reached, if it somehow does get here an error instead of trying to force the grab doesn't make much sense and it should throw instead.